src/hotspot/share/prims/jvm.cpp

```diff
@@ -2127,10 +2126,15 @@ JVM_ENTRY(jobjectArray, JVM_GetPermittedSubclasses(JNIEnv* env, jclass current))
   Klass* c = java_lang_Class::as_Klass(mirror);
   assert(c->is_instance_klass(), "must be");
   InstanceKlass* ik = InstanceKlass::cast(c);
+  log_trace(class, sealed)("Calling GetPermittedSubclasses for %s type %s",
```

Comment:
I don't know why this didn't complain about a missing ResourceMark, but can you add one just before line 2129 and it'll clean up any potential resource area uses for any external_name() calls.  They're not that expensive so one would do without conditionalization.

Reply:
I added the ResourceMark in the latest commit. Thanks for pointing out that it was missing!
Harold